### PR TITLE
🌱 Harden /health Netlify Function: CORS allowlist, canonical branding, tests

### LINKED ISSUE #17567: Quantum dashboard not visible on console.kubestellar.io demo site
## Summary

The **Quantum Demo** dashboard does not appear in the left-rail sidebar at https://console.kubestellar.io, even though demo mode is forced on that hostname and the four quantum cards already render correctly in their forced-demo state.

## Repro

1. Visit https://console.kubestellar.io
2. Open the left sidebar (default + customized views)
3. Observe: no **Quantum Demo** entry. Direct navigation to https://console.kubestellar.io/quantum still works (the route is registered), but a user cannot discover the dashboard from the UI.

## Root cause

Three things compound:

1. **`quantum` lives in `DISCOVERABLE_DASHBOARDS`, not `DEFAULT_PRIMARY_NAV`** — see [`web/src/hooks/useSidebarConfig.ts:79-101`](https://github.com/kubestellar/console/blob/main/web/src/hooks/useSidebarConfig.ts#L79-L101). Discoverable items are intentionally NOT shown in the sidebar by default; the comment on lines 75-77 confirms this. They appear only if (a) the user adds them via the sidebar customizer, or (b) the server promotes them via the `enabled_dashboards` field on `/health`.

2. **The Go backend already promotes quantum on self-hosted consoles.** [`pkg/api/routes_health.go:33-104`](https://github.com/kubestellar/console/blob/main/pkg/api/routes_health.go#L33-L104) returns `enabled_dashboards`, sourced from the kubestellar project preset in [`pkg/api/projects.go:9-19`](https://github.com/kubestellar/console/blob/main/pkg/api/projects.go#L9-L19), which DOES include `\"quantum\"`. On a self-hosted (non-Netlify) console, quantum is promoted into the sidebar correctly by `applyDashboardFilter` ([useSidebarConfig.ts:272-301](https://github.com/kubestellar/console/blob/main/web/src/hooks/useSidebarConfig.ts#L272-L301)).

3. **There is no `/health` Netlify Function and no `/health` redirect in `netlify.toml`.** On console.kubestellar.io, `fetch('/health')` falls through to the SPA's `index.html`. The `await resp.json()` call at [`useSidebarConfig.ts:308`](https://github.com/kubestellar/console/blob/main/web/src/hooks/useSidebarConfig.ts#L308) throws on the HTML body, the silent `catch {}` at [line 334](https://github.com/kubestellar/console/blob/main/web/src/hooks/useSidebarConfig.ts#L334) swallows it, `enabledDashboardIds` stays `null`, and the promotion path never runs.

Verifying directly:

\`\`\`
$ curl -sI https://console.kubestellar.io/health | grep content-type
content-type: text/html; charset=utf-8
\`\`\`

## Proposed fix

Add `web/netlify/functions/health.mts` with a JSON shape parity to `pkg/api/routes_health.go`, hard-coded for the hosted-demo case:

- `project: \"kubestellar\"` (drives branding flags + enabled_dashboards lookup)
- `workloads.quantum_kc_demo_available: false` (keeps the four quantum cards on demo data via `isQuantumForcedToDemo()` — desired hosted behavior)
- `enabled_dashboards`: copy of the kubestellar preset from `pkg/api/projects.go`

Plus a `/health` redirect in `netlify.toml` and an MSW passthrough/handler so demo-mode dev/test stays consistent.

PR follows.

## Related

- Existing E2E coverage for the four quantum cards in forced-demo state: `web/e2e/user-flows/quantum-demo.spec.ts` (PR #17427)

### PR DESCRIPTION
**Pull request by KPRoche**
## Summary

Fixes #17567 — the Quantum Demo dashboard was invisible in the sidebar at https://console.kubestellar.io despite the route working and the cards rendering correctly in forced-demo mode.

- **Add `web/netlify/functions/health.mts`** — mirrors `pkg/api/routes_health.go`'s response with values appropriate for the hosted demo (`project: \"kubestellar\"`, `quantum_kc_demo_available: false`, full kubestellar `enabled_dashboards` preset). Without this, `fetch('/health')` fell through to the SPA, `resp.json()` threw on HTML, and the sidebar's `applyDashboardFilter` promotion path never ran (`useSidebarConfig.ts:307-336`).
- **Add `/health` redirect in `netlify.toml`** so requests reach the new function instead of the SPA fallback.
- **Add MSW handler for `/health`** in `web/src/mocks/handlers.endpoints.ts` (separate endpoint from `/api/health`) so demo-mode dev/test surfaces the same dashboards locally.
- **Update CLAUDE.md route-parity table** with a sync-with-`pkg/api/projects.go` note on the new entry.

The `enabled_dashboards` array is verified order-preserving identical (33 entries) across `pkg/api/projects.go`, `health.mts`, and the MSW handler.

## Test plan

- [ ] `curl -s https://deploy-preview-NNN--kubestellar-console.netlify.app/health | jq '.enabled_dashboards | index(\"quantum\")'` → non-null integer (preview deploy)
- [ ] Visit deploy-preview URL, confirm **Quantum Demo** appears in the left-rail sidebar
- [ ] Click Quantum Demo → all four cards render with the yellow Demo outline + badge (forced-demo via `isQuantumForcedToDemo()`)
- [ ] ControlPanel shows \"Not configured\" badge (covered by existing E2E `web/e2e/user-flows/quantum-demo.spec.ts` from PR #17427)
- [ ] After merge, repeat the curl + visual check against https://console.kubestellar.io
- [ ] Sidebar on a self-hosted (non-Netlify) console is unchanged — Go backend still serves `/health` directly

## Notes

- No backend code changes — Netlify-side parity only.
- The dashboard preset is duplicated between Go (`pkg/api/projects.go`) and the new `.mts` file. A code comment in each spot flags the sync requirement; a generated single-source approach felt over-engineered for a list that changes a few times a year.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: web/src/mocks/handlers.endpoints.ts [hunk 1]
```diff
@@ -133,6 +133,41 @@ export function createHandlers() {
...
+        'insights', 'ai-ml', 'ai-agents', 'acmm', 'ci-cd',
+        'multi-tenancy', 'alerts', 'arcade', 'quantum',
+        'llm-d-benchmarks', 'gpu-reservations',
+        'compute', 'security', 'storage', 'network', 'events',
+        'workloads', 'operators', 'nodes', 'deployments', 'pods',
+        'services', 'helm', 'logs', 'data-compliance', 'cost',
+        'gitops', 'gpu',
+      ],
+    })
+  }),
```
[the remark this answers is not in the document]
Comment (KPRoche) [reply web/src/mocks/handlers.endpoints.ts]: Addressed in `be50c7555`. Behavior was already identical because [`mergeBranding`](https://github.com/kubestellar/console/blob/be50c7555/web/src/lib/branding.ts#L90-L102) explicitly skips empty strings and falls back to `DEFAULT_BRANDING` (KubeStellar values), so an MSW response missing `appName` and a Netlify response sending `appName: ""` both ended up showing "KubeStellar Console" in the UI. But you're right that the differing shapes were a foot-gun, so the MSW handler now mirrors `health.mts` field-for-field — and the parity test ([web/netlify/functions/__tests__/health.test.ts](https://github.com/kubestellar/console/blob/be50c7555/web/netlify/functions/__tests__/health.test.ts)) covers branding too.

File: web/netlify/functions/health.mts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  "compute", "security", "storage", "network", "events",
+  "workloads", "operators", "nodes", "deployments", "pods",
+  "services", "helm", "logs", "data-compliance", "cost",
+  "gitops", "gpu",
+];
+
+export default async (req: Request): Promise<Response> => {
+  if (req.method === "OPTIONS") {
+    return handlePreflight(req, CORS_OPTIONS);
+  }
```
[the remark this answers is not in the document]
Comment (KPRoche) [reply web/netlify/functions/health.mts:74]: Addressed in `be50c7555`. The handler now returns `405` with `Allow: GET, OPTIONS` for non-GET/non-OPTIONS requests, with a test in `health.test.ts:73-79` covering it.

File: web/netlify/functions/health.mts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const KUBESTELLAR_ENABLED_DASHBOARDS = [
+  "dashboard", "clusters", "cluster-admin", "compliance", "deploy",
+  "insights", "ai-ml", "ai-agents", "acmm", "ci-cd",
+  "multi-tenancy", "alerts", "arcade", "quantum",
+  "llm-d-benchmarks", "gpu-reservations",
+  "compute", "security", "storage", "network", "events",
+  "workloads", "operators", "nodes", "deployments", "pods",
+  "services", "helm", "logs", "data-compliance", "cost",
+  "gitops", "gpu",
+];
```
[the remark this answers is not in the document]
Comment (KPRoche) [reply web/netlify/functions/health.mts]: Good idea on the surface, but the bundle boundary makes it costly here: the Netlify Function bundle (esbuild) and the Vite frontend bundle are intentionally isolated — no existing function imports from `src/`, and `web/tsconfig.json` only includes `src` (not `netlify/`). Reaching across has bitten us before with `import.meta.env.DEV` accidentally landing in Node-side code. So extracting `KUBESTELLAR_ENABLED_DASHBOARDS` to a shared module would force one bundle to import from the other, which is unprecedented in this repo.

What enforces the rule instead is the parity test in [`web/netlify/functions/__tests__/health.test.ts`](https://github.com/kubestellar/console/blob/be50c7555/web/netlify/functions/__tests__/health.test.ts) — three tests that read `pkg/api/projects.go`, `health.mts`, and `handlers.endpoints.ts` directly and assert order-preserving equality across all three copies. Verified by intentionally renaming `"quantum"` → `"quantumDRIFT"`: 3 of 9 tests fail; restoring makes them pass. So drift fails CI rather than relying on a "keep these in sync" comment.

(Also note the Go list will always be a separate copy regardless — Go can't import from a TS module. The parity test handles that side too.)

**Comment by KPRoche** [comment]
Update: parallel auto-scanner PR #17569 caught a real defect — the original commit (`927be57e7`) added a NEW `/health` MSW handler in `handlers.endpoints.ts` while a stub already existed in `handlers.platform.ts:191-195`. MSW picks the first-registered handler, so the duplicate shadowed the canonical one.

Fixed in `82c86c5fa`:
- Remove the duplicate from `handlers.endpoints.ts`; leave a pointer comment to the canonical location.
- Update the existing `handlers.platform.ts` handler to return the full body shape (matching `health.mts`).
- Update the parity test to read `handlers.platform.ts`.
- Add a regression test that fails if a duplicate `/health` handler is ever re-introduced.

Verified: 10 tests pass; regression test fires on intentional duplicate, restored after.

Posted a comparison on #17569 asking maintainers to close it in favor of this PR (which has the test coverage + the proper [`_shared/cors.ts`](https://github.com/kubestellar/console/blob/main/web/netlify/functions/_shared/cors.ts) allowlist usage instead of `Access-Control-Allow-Origin: *`).
